src/network/netdev/batadv.h

```diff
@@ -0,0 +1,41 @@
+#pragma once
+
+/* SPDX-License-Identifier: LGPL-2.1+ */
+
+#include <linux/batman_adv.h>
+
+#include "conf-parser.h"
+#include "netdev/netdev.h"
+
+
+#define BATADV_GENL_NAME "batadv"
+#define BATADV_GENL_VERSION 1
+
+typedef struct Batadv {
+        NetDev meta;
+
+        uint8_t gateway_mode;
+        uint32_t gateway_bandwidth_down;
+        uint32_t gateway_bandwidth_up;
+        uint8_t hop_penalty;
+        usec_t originator_interval;
+        bool aggregation;
+        bool bridge_loop_avoidance;
+        bool distributed_arp_table;
+        bool fragmentation;
+} BatmanAdvanced;
+
+typedef enum BatadvGatewayModes {
+        BATADV_GATEWAY_MODE_OFF,
+        BATADV_GATEWAY_MODE_CLIENT,
+        BATADV_GATEWAY_MODE_SERVER,
+} BatadvGatewayModes;
```

Comment:
- Please move this definition above the definition of `struct BatmanAdvanced`.
- Please specify each value explicitly, and define _MAX and _INVALID:
```
typedef enum BatadvGatewayModes {
        BATADV_GATEWAY_MODE_OFF    = BATADV_GW_MODE_OFF,
        BATADV_GATEWAY_MODE_CLIENT = BATADV_GW_MODE_CLIENT,
        BATADV_GATEWAY_MODE_SERVER = BATADV_GW_MODE_SERVER,
        _BATADV_GATEWAY_MODE_MAX,
        _BATADV_GATEWAY_MODE_INVALID = -1,
} BatadvGatewayModes;
```